specs/sphinx-module-factory.md

```diff
@@ -0,0 +1,104 @@
+# `SphinxModuleFactory` Contract Specification
+
+The `SphinxModuleFactory` deploys `SphinxModule` proxy contracts at deterministic addresses and enables them within Gnosis Safe contracts.
+
+It uses the [EIP-1167](https://eips.ethereum.org/EIPS/eip-1167) standard to reduce the cost of deploying `SphinxModule` contracts. Instead of deploying a new `SphinxModule` contract for every Gnosis Safe, it deploys a minimal EIP-1167 proxy that delegates all calls to a single `SphinxModule` implementation contract. The `SphinxModuleFactory` deploys the `SphinxModule` implementation inside its constructor.
+
+**Vocabulary notes**:
+* A _SphinxModule_ is an EIP-1167 proxy that delegates calls to a `SphinxModule` implementation contract. If we're referring to the `SphinxModule` implementation contract itself, we'll call it the `SphinxModule` implementation.
+* A _Gnosis Safe_ is a Gnosis Safe proxy contract that delegates calls to a Gnosis Safe implementation.
```

Comment:
Nit: It would probably be clearer to mirror the naming scheme that Safe uses. 

They have the following
- GnosisSafe: The Safe implementation.
- GnosisSafeProxy: The minimal proxy used to make deploying cheap.
- GnosisSafeProxyFactory: The factory used to deploy new Safes behind `GnosisSafeProxy` contracts.

So for us, I would use:
- SphinxModule: The implementation.
- SphinxModuleProxy: The proxy.
- GnosisModuleProxyFactory: The factory used to deploy new Safes behind `SphinxModuleProxy`.

Obviously, in our case we don't have our own implementation for `SphinxModuleProxy`. Instead, we use the OpenZeppelin Clones library (good call) but I would still use the terminology.